third_party/flux/flux_setup.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flux
+
+import (
+	"context"
+	"fmt"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+var manager *Manager
+
+const NoFluxInstallationFoundMsg = "flux needs to be installed within a cluster first"
+
+func InstallFlux() env.Func {
```

Comment:
Do you think it would be a good idea to take the namespace as an optional argument here so that one can easily install this on a non standard `flux-system` namespace ?